app/bundles/ReportBundle/Model/CsvExporter.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * @copyright   2014 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\ReportBundle\Model;
+
+use Mautic\CoreBundle\Templating\Helper\FormatterHelper;
+use Mautic\ReportBundle\Crate\ReportDataResult;
+
+/**
+ * Class CsvExporter.
+ */
+class CsvExporter
+{
+    /**
+     * @var FormatterHelper
+     */
+    protected $formatterHelper;
+
+    public function __construct(FormatterHelper $formatterHelper)
+    {
+        $this->formatterHelper = $formatterHelper;
+    }
+
+    public function export(ReportDataResult $reportDataResult, $handle, $page = 1)
+    {
+        if ($page === 1) {
+            $this->putHeader($reportDataResult, $handle);
```

Comment:
The file is completely empty if there is no data. Can we still add the headers even if empty?

Reply:
This behaviour is same like before this PR.
I do not have columns if no data provided. I could put all columns as a header there, but that does not make sense I think.
I can refactor `getReportData` method, but it is pretty huge and it takes time.